bin/ls

```diff
@@ -41,6 +41,17 @@ my @perms = qw(--- --x -w- -wx r-- r-x rw- rwx);
 my @ftype = qw(. p c ? d ? b ? - ? l ? s ? ? ?);
 $ftype[0] = '';
 
+sub get_columns {
+  eval {
+    require IO::Tty;
```

Comment:
IO::Tty doesn't work on Windows, and on linux and friends I think it's too much. Here are some things that will get the columns:

    stty size  | cut -d' ' -f 2

    tput cols

On Windows with powershell (with Perl post processing):

    powershell -command "&{(get-host).ui.rawui.WindowSize;}"